tls/s2n_x509_validator.c

```diff
@@ -549,3 +550,64 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_stapled_ocsp_response(
     return ret_val;
 #endif /* S2N_OCSP_STAPLING_SUPPORTED */
 }
+
+DEFINE_POINTER_CLEANUP_FUNC(STACK_OF(X509)*, wipe_cert_chain);
+
+s2n_cert_validation_code s2n_x509_validator_validate_certificate_signatures(struct s2n_x509_validator *validator,  struct s2n_connection *conn)
+{
+    if (validator->skip_cert_validation || !conn->config->security_policy->certificate_signature_preferences) {
+        return S2N_CERT_OK;
+    }
+    S2N_ERROR_IF((validator->state != VALIDATED && validator->state != OCSP_VALIDATED), S2N_ERR_INVALID_STATE);
+    
+    DEFER_CLEANUP(STACK_OF(X509) *validated_chain = X509_STORE_CTX_get1_chain(validator->store_ctx), wipe_cert_chain_pointer);
+
+    const int certs_in_chain = sk_X509_num(validated_chain);
+
+    /* Do not validate the root certificate */
+    int certs_to_validate = certs_in_chain - 1;
```

Comment:
This may be better as an `unsigned` . I'm not sure what guarantees `sk_x509_num` gives, but there should never be less than 0 certs.